src/string_converter_filter.cpp

```diff
@@ -368,6 +371,19 @@ InternalValue StringConverter::Filter(const InternalValue& baseVal, RenderContex
             return str;
         });
         break;
+    case CenterMode:
+        result = ApplyStringConverter(baseVal, [this, &context](auto srcStr) -> TargetString {
+            auto width = ConvertToInt(this->GetArgumentValue("width", context));
+            auto str = sv_to_string(srcStr);
```

Comment:
I think, it should be additional trimming added at this point. Now you doesn't count the spaces which already present at the beginning or at the end of the string,

Reply:
Are you sure about that?

I tested this and it seems that is is not trimming those whitespaces:

```python
from jinja2 import Environment

HTML = """
{{ 'x' | center(width=5) }}
{{ '  x' | center(width=5) }}
"""

def print_html_doc():
    print (Environment().from_string(HTML).render()
print_html_doc()
```

Reply:
Ok. You're right. Thank you!